src/lbaf/IO/lbsVTDataWriter.py

```diff
@@ -58,32 +56,66 @@ def __create_object_entries(self, rank_id, objects):
             "time": o.get_load()}
             for o in objects]
 
-    def json_writer(self, rank: Rank) -> str:
+    def _json_writer(self, rank_phases_double) -> str:
+        """ Write one JSON per rank for list of phase instances."""
+        # Unpack received double
+        r_id, r_phases = rank_phases_double
+
         # Create file name for current rank
-        file_name = f"{self.__file_stem}.{rank.get_id()}.{self.__suffix}"
-        if self.__output_dir is not None:
-            file_name = os.path.join(self.__output_dir, file_name)
+        file_name = f"{self.__file_stem}.{r_id}.{self.__extension}"
+        self.__logger.debug(f"Writing {file_name}")
 
         # Initialize output dict
-        phase_data = {"id": self.__phase.get_id()}
-        r_id = rank.get_id()
         output = {
             "metadata": {
                 "type": "LBDatafile",
                 "rank": r_id},
-            "phases": [phase_data]}
-
-        # Create list of objects descriptions
-        tasks = self.__create_object_entries(
-            r_id, rank.get_migratable_objects())
-        tasks += self.__create_object_entries(
-            r_id, rank.get_sentinel_objects())
-        phase_data["tasks"] = tasks
-
-        # Write file and return its name
-        json_str = json.dumps(output, separators=(',', ':'))
-        compressed_str = brotli.compress(
-            string=json_str.encode("utf-8"), mode=brotli.MODE_TEXT)
-        with open(file_name, "wb") as compr_json_file:
-            compr_json_file.write(compressed_str)
+            "phases": []}
```

Comment:
Again, not everything that appears in an input file is being output here. There can be more content to the metadata than is shown.

Reply:
Following our discussion of 05/03 I believe this is is resolved